### ISSUE TITLE
ZipArchive.GetEntry case sensitive, even on case insensitive OS

### ISSUE THREAD

**Issue by EskeRahn** [dotnet/runtime#48267]
ZipArchive.GetEntry is always case sensitive.
In a OS (like windows) that is insensitive, the input cannot be expected to be provided case sensitive.

Similar to #1553 this is OS dependent.

A new option, to force it to be case **in**sensitive would solve it. An easy way could be to add a bit to ZipArchiveMode

Optimally throwing an exception if the provided name matches more than one entry case **in**sensitive.
(Of course a workaround is just browsing through all entries, and check manually case insensitive)

**Comment by danmoseley** [comment]
What would this flag do if there were two file name in the zip that differed only in casing?

**Comment by huoyaoyuan** [comment]
Note that *Windows isn't case insensitive in deep*. Case sensitivity can be turned on.

**Comment by EskeRahn** [comment]
Thanks for the quick reactions.

> What would this flag do if there were two file name in the zip that differed only in casing?

As suggested then the optimal would be to have an Exception, telling that an ambiguous name have been requested. So people would either have to access the archive case-sensitive OR to access by index (or foreach).

IF you also added a method returning a **collection**, for usage with **wildcards**, That could also return all variants matching, if case sensitivity was turned off.



> Note that _Windows isn't case insensitive in deep_. Case sensitivity can be turned on.

I'm aware, but frankly never came across anyone exploiting that. So at the least an option would be nice.

**Comment by danmoseley** [comment]
Thanks for the suggestion. Could you update your top post to match the API proposal template? There is a special new issue template for it.

**Comment by EskeRahn** [comment]
ADD:
Just noticed by an accident, that it is allowed today to add several entries with identical names, so the duality looking op a single name would be the same as an case-insensitive lookup of entries that differed by case only

**Comment by karelz** [comment]
Triage: Sorry for the XXL delay. We are now reviewing all issues in Compression backlog.
The proposed change to default (on Windows only) would be a technical breaking change. Given that the demand is low (1 ask in last 4.5 years), we do not think it is worth it.
Especially, given that there is a simple workaround - enumerate all entries and do name comparison manually.

Therefore, we will close the issue. Let us know if we missed anything.

**Comment by EskeRahn** [comment]
> would be a technical breaking change.

No! I suggested it as an OPTION, it could still be cassensitive by default - even on windows.

But indeed there has not been much support of this idea here *LOL*
